libs/kotlinlib/src/mill/kotlinlib/ksp/KspWorkerModule.scala

```diff
@@ -0,0 +1,36 @@
+package mill.kotlinlib.ksp
+
+import mill.api.{Discover, ExternalModule, PathRef}
+import mill.util.Jvm
+
+@mill.api.experimental
+trait KspWorkerModule extends ExternalModule {
+  def runKsp(logLevel: String, workerClasspath: Seq[PathRef], kspArgs: Seq[String]): Unit
+}
+
+@mill.api.experimental
+object KspWorkerModule extends KspWorkerModule {
+
+  def runKsp(logLevel: String, workerClasspath: Seq[PathRef], kspArgs: Seq[String]): Unit = {
+
+    val kspClassLoader = Jvm.createClassLoader(
+      workerClasspath.map(_.path),
+      getClass.getClassLoader
+    )
```

Comment:
Does that mean, we re-create the classloader for every invocation? Instead, we should share the classloader, which should improve performance for repetitive invocations (watch-mode or multi-module use).

Reply:
how would it detect classpath changes? can this be achieved by creating a classloader in a task? or could you point me to an example?

Reply:
Lets first name the various classloaders and their lifecycles and then find the best implementation.